docs/codeql/codeql-cli/about-codeql-packs.rst

```diff
@@ -59,46 +62,254 @@ The following properties are supported in ``qlpack.yml`` files.
+     - Metadata that will be displayed on the packaging search page in the packages section of the account that the CodeQL pack is published to. For a list of allowed licenses, see `SPDX License List <https://spdx.org/licenses/>`__ in the SPDX Specification.
    * - ``description``
-     - ``Human-readable description of the contents of the CodeQL pack.``
+     - .. code-block:: yaml
+
+        description: Human-readable description of the contents of the CodeQL pack.
+
      - All packs
      - Metadata that will be displayed on the packaging search page in the packages section of the account that the CodeQL pack is published to.
+   * - ``libraryPathDependencies``
+     - .. code-block:: yaml
+
+        libraryPathDependencies: codeql/javascript-all
+
+     - Optional, deprecated
+     - Use the ``dependencies`` property instead. The names of any CodeQL packs that this CodeQL pack depends on, as a sequence. This gives the pack access to any libraries, database schema, and query suites defined in the dependency.
+
+.. _about-codeql-pack-lock:
+
+About ``codeql-pack.lock.yml`` files
+------------------------------------
+
+``codeql-pack.lock.yml`` files store the versions of the resolved transitive dependencies of a CodeQL pack. This file is created by the ``codeql pack install`` if it does not already exist and should be added to your version control system. The ``dependencies`` section of the ``qlpack.yml`` file contains version ranges that are compatible with the pack. The ``codeql-pack.lock.yml`` file locks the versions to precise dependencies. This ensures that anyone running ``codeql pack install`` on this the pack will always retrieve the same versions of dependencies even if newer compatible versions exist. Only query packs can have ``codeql-pack.lock.yml`` files. This is because library packs are not compiled independently of query packs. Without being compiled, there is no need to specify precise versions of their dependencies.
+
+For example, if a ``qlpack.yml`` file contains the following dependencies:
+
+.. code-block:: yaml
+
+   dependencies:
+     codeql/cpp-all: ^0.1.2
+     my-user/my-lib: ^0.2.3
+     other-dependency/from-source: "*"
+
+The ``codeql-pack.lock.yml`` file will contain something like the following:
+
+.. code-block:: yaml
+
+   dependencies:
+     codeql/cpp-all:
+       version: 0.1.4
+     my-user/my-lib:
+       version: 0.2.4
+     my-user/transitive-dependency:
+       version: 1.2.4
+
+The ``codeql/cpp-all`` dependency is locked to version 0.1.4. The ``my-user/my-lib`` dependency is locked to version 0.2.4. The ``my-user/transitive-dependency``, which is a transitive dependency and is not specified in the ``qlpack.yml`` file, is locked to version 1.2.4. The ``other-dependency/from-source`` is absent from the lock file since it is resolved from source. This dependency must be available in the same CodeQL workspace as the pack.
```

Comment:
Do we document what it means for a pack to be resolved from source?

Reply:
We aren't yet. That will happen in a later change, in a new article that describes how query resolution occurs.

We should add a TODO here to remind ourselves to add a link to that article here when we create it.